docs/getting-started-guides/mesos.md

```diff
@@ -0,0 +1,287 @@
+ 0:00:03    R    67.62 MB  0.5   105.66   none  root bda74154-7a61-11e4-8b8c-42010a863922
+ 0:00:03    R    68.20 MB  0.75  106.57   none  root bda6e97f-7a61-11e4-8b8c-42010a863922
+ 0:00:03    R    68.20 MB  0.75  106.57   none  root b0e206f1-7a61-11e4-8b8c-42010a863922
+```
+
+Determine the internal IP address of the frontend [service portal][8]:
+
+```bash
+$ bin/kubectl get services
+NAME            LABELS                                    SELECTOR            IP             PORT
+kubernetes      component=apiserver,provider=kubernetes   <none>              10.10.10.2     443
+kubernetes-ro   component=apiserver,provider=kubernetes   <none>              10.10.10.1     80
+redismaster     <none>                                    name=redis-master   10.10.10.49    10000
+redisslave      name=redisslave                           name=redisslave     10.10.10.109   10001
+frontend        <none>                                    name=frontend       10.10.10.149   9998
+```
+
+Interact with the frontend application via curl:
+
+```bash
+$ curl http://${frontend_service_ip_address}:9998/index.php?cmd=get\&key=messages
+```
+
+Or via the Redis CLI:
+
+```bash
+$ sudo apt-get install redis-tools
+$ redis-cli -h ${redis_master_service_ip_address} -p 10000
+10.233.254.108:10000> dump messages
+"\x00\x06,world\x06\x00\xc9\x82\x8eHj\xe5\xd1\x12"
+```
+
+Tail the logs of the kubelet-executor:
+
+```bash
+$ mesos tail -f c7315c47-7a61-11e4-8b8c-42010a863922 stderr
+I1202 20:34:38.749340 03297 log.go:151] GET /podInfo?podID=c7312b26-7a61-11e4-8b8c-42010a863922&podNamespace=default: (3.71265ms) 200
+I1202 20:34:38.772047 03297 log.go:151] GET /podInfo?podID=bda7385a-7a61-11e4-8b8c-42010a863922&podNamespace=default: (3.785459ms) 200
+...
+```
+
+Or interact with the frontend application via your browser, in 2 steps:
+
+First, open the firewall on the master machine.
+
+```bash
+$ # determine the internal port for the frontend service portal
+$ sudo iptables-save|grep -e frontend  # -- port 36336 in this case
+-A KUBE-PORTALS-CONTAINER -d 10.10.10.149/32 -p tcp -m comment --comment frontend -m tcp --dport 9998 -j DNAT --to-destination 10.22.183.23:36336
+-A KUBE-PORTALS-CONTAINER -d 10.22.183.23/32 -p tcp -m comment --comment frontend -m tcp --dport 9998 -j DNAT --to-destination 10.22.183.23:36336
+-A KUBE-PORTALS-HOST -d 10.10.10.149/32 -p tcp -m comment --comment frontend -m tcp --dport 9998 -j DNAT --to-destination 10.22.183.23:36336
+-A KUBE-PORTALS-HOST -d 10.22.183.23/32 -p tcp -m comment --comment frontend -m tcp --dport 9998 -j DNAT --to-destination 10.22.183.23:36336
+
+$ # open up access to the internal port for the frontend service portal
+$ sudo iptables -A INPUT -i eth0 -p tcp -m state --state NEW,ESTABLISHED -m tcp \
+  --dport ${internal_frontend_service_port} -j ACCEPT
+```
+
+Next, add a firewall rule in Google Cloud Platform Console / Networking:
+
+![Google Cloud Platform firewall configuration][9]
+
```

Comment:
Yep that appears to work too. If executed from the command line of the GCE
VM then we'd need to have the user authenticate first. I just tried it and
got a warning:

Your credentials may be visible to others with access to this
virtual machine. Are you sure you want to authenticate with
your personal account?

I'm not sure if that would inhibit people. If they have gcloud installed on
their local machine then it's (hopefully) more secure but that adds a
dependency to the tutorial.

Thoughts?

On Mon, Mar 16, 2015 at 3:56 PM, Eric Tune <notifications@github.com> wrote:

> In docs/getting-started-guides/mesos.md
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5442#discussion_r26521270>
> :
>
> > +$ # determine the internal port for the frontend service portal
> > +$ sudo iptables-save|grep -e frontend  # -- port 36336 in this case
> > +-A KUBE-PORTALS-CONTAINER -d 10.10.10.149/32 -p tcp -m comment --comment frontend -m tcp --dport 9998 -j DNAT --to-destination 10.22.183.23:36336
> > +-A KUBE-PORTALS-CONTAINER -d 10.22.183.23/32 -p tcp -m comment --comment frontend -m tcp --dport 9998 -j DNAT --to-destination 10.22.183.23:36336
> > +-A KUBE-PORTALS-HOST -d 10.10.10.149/32 -p tcp -m comment --comment frontend -m tcp --dport 9998 -j DNAT --to-destination 10.22.183.23:36336
> > +-A KUBE-PORTALS-HOST -d 10.22.183.23/32 -p tcp -m comment --comment frontend -m tcp --dport 9998 -j DNAT --to-destination 10.22.183.23:36336
> > +
> > +$ # open up access to the internal port for the frontend service portal
> > +$ sudo iptables -A INPUT -i eth0 -p tcp -m state --state NEW,ESTABLISHED -m tcp \
> > +  --dport ${internal_frontend_service_port} -j ACCEPT
> > +```
> > +
> > +Next, add a firewall rule in Google Cloud Platform Console / Networking:
> > +
> > +![Google Cloud Platform firewall configuration][9]
> > +
>
> does something like this work:
>
> gcloud compute firewall-rules create frontend-service --source-ranges
> 0.0.0.0/0 --allow tcp:9998
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5442/files#r26521270>
> .
>



-- 
James DeFelice
585.241.9488 (voice)
650.649.6071 (fax)